source/extensions/clusters/aggregate_retry/lb_context.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "source/common/upstream/upstream_impl.h"
+#include "source/extensions/load_balancing_policies/common/load_balancer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace AggregateRetry {
+
+// AggregateRetryLoadBalancerContext wraps the load balancer context to provide retry-aware
+// cluster selection while delegating most operations to the underlying context.
+class AggregateRetryLoadBalancerContext : public Upstream::LoadBalancerContextBase {
+public:
+  AggregateRetryLoadBalancerContext(Upstream::LoadBalancerContext* context,
+                                    size_t selected_cluster_index)
+      : selected_cluster_index_(selected_cluster_index) {
+    if (context == nullptr) {
```

Comment:
Do you need this wrapper just for tests?

Reply:
We need it outside of tests as well since we need to pass both the original `LoadBalancerContext` (for headers, metadata, etc.) as well as the selected cluster index (for tracking/debugging). This wrapper combines both these.